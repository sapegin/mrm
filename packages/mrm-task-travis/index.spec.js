jest.mock('fs');
jest.mock('git-username');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));

const { getTaskOptions } = require('mrm');
const vol = require('memfs').vol;
const task = require('./index');

const console$log = console.log;

const stringify = o => JSON.stringify(o, null, '  ');

const packageJson = stringify({
	name: 'unicorn',
	engines: {
		node: 4,
	},
	scripts: {
		test: 'jest',
	},
});

beforeEach(() => {
	console.log = jest.fn();
});

afterEach(() => {
	vol.reset();
	console.log = console$log;
});

it('should add Travis CI', async () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(await getTaskOptions(task));

	expect(vol.toJSON()).toMatchSnapshot();
});

it('should add Travis CI, with yarn.lock', async () => {
	vol.fromJSON({
		'/yarn.lock': '',
		'/package.json': packageJson,
	});

	task(await getTaskOptions(task));

	expect(vol.toJSON()).toMatchSnapshot();
});

it('should add latest Node version if engines field is not defined', async () => {
	vol.fromJSON({
		'/package.json': stringify({
			name: 'unicorn',
			scripts: {
				test: 'jest',
			},
		}),
	});

	task(await getTaskOptions(task));

	expect(vol.toJSON()['/.travis.yml']).toMatchSnapshot();
});

it('should add latest Node version if engines field is not defined', async () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(await getTaskOptions(task, false, { maxNode: 14 }));

	expect(vol.toJSON()['/.travis.yml']).toMatchSnapshot();
});

it('should add a badge to the Readme', async () => {
	vol.fromJSON({
		'/package.json': packageJson,
		'/Readme.md': '# Unicorn',
	});

	task(await getTaskOptions(task));

	expect(vol.toJSON()['/Readme.md']).toMatchSnapshot();
});
