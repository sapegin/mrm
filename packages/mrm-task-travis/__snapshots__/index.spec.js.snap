// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add Travis CI 1`] = `
Object {
  "/.travis.yml": "language: node_js
cache:
  directories:
    - node_modules
node_js:
  - 4
  - 6
  - 8
  - 10
  - 12
",
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"engines\\": {
    \\"node\\": 4
  },
  \\"scripts\\": {
    \\"test\\": \\"jest\\"
  }
}",
}
`;

exports[`should add Travis CI, with yarn.lock 1`] = `
Object {
  "/.travis.yml": "language: node_js
cache:
  directories:
    - node_modules
  yarn: true
node_js:
  - 4
  - 6
  - 8
  - 10
  - 12
",
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"engines\\": {
    \\"node\\": 4
  },
  \\"scripts\\": {
    \\"test\\": \\"jest\\"
  }
}",
  "/yarn.lock": "",
}
`;

exports[`should add a badge to the Readme 1`] = `
"# Unicorn

[![Build Status](https://travis-ci.org/mrm/unicorn/unicorn.svg)](https://travis-ci.org/mrm/unicorn/unicorn)"
`;

exports[`should add latest Node version if engines field is not defined 1`] = `
"language: node_js
cache:
  directories:
    - node_modules
node_js:
  - 12
"
`;

exports[`should add latest Node version if engines field is not defined 2`] = `
"language: node_js
cache:
  directories:
    - node_modules
node_js:
  - 4
  - 6
  - 8
  - 10
  - 12
  - 14
"
`;
